ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java

```diff
@@ -544,6 +543,7 @@ public static Bytes32 get_randao_mix(BeaconState state, long epoch) {
         epoch <= get_current_epoch(state), "checkArgument threw and exception in get_randao_mix()");
     long index = epoch % LATEST_RANDAO_MIXES_LENGTH;
     List<Bytes32> randao_mixes = state.getLatest_randao_mixes();
+    if (index == -1) index = 0;
```

Comment:
This feels like we are hiding a problem rather than fixing one.  Could the test be modified rather than the state transition code? If this isn't in the spec, we should be skeptical about adding it.

Reply:
It's due to GENESIS_SLOT being set to 0, and get_randao_mix indexes -1 (see description).